app/Models/AssetModel.php

```diff
@@ -150,6 +150,11 @@ public function fieldset()
     {
         return $this->belongsTo(\App\Models\CustomFieldset::class, 'fieldset_id');
     }
+   
+    public function customFields()
+    {
+       return $this->fieldset()->first()->fields(); 
```

Comment:
I think we can use a [`hasManyThrough`](https://laravel.com/docs/8.x/eloquent-relationships#has-many-through) here so this relationship can be eager loaded and cut down on queries.

Reply:
If I'm remembering correctly the reason I didn't do this originally is because `hasManyThrough` doesn't natively support going through a pivot, and it gets a little heady so I'd stuck with 'good enough' for now.

Reply:
Gotcha!